lib/tree.rb

```diff
@@ -1,38 +1,60 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+RANDOM = Random.new;
+
+class Tree
+  attr_reader :height, :age, :apples, :alive
 
   def initialize
+    @age = 0
+    @height = 1
+    @apples = []
+    @alive = true
   end
 
   def age!
+    @age += 1
+    if @alive
```

Comment:
i may be misunderstanding you but the preference is w/o `self` in the case of ivars. they're referring to the same instance method in this case anyway.

unrelated: you'll see `self` used for class methods (e.g. ActiveRecord querying) 

```ruby
def self.klass
  p self
end
```

or 

```ruby
class << self
  def klass
    self
  end
end
```